### PR TITLE
🏃 Run CAPD verify scripts in CI

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Signed-off-by: Ashish Amarnath <ashisham@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: test/infrastructure/docker/hack/verify-spelling.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ URL="https://github.com/client9/misspell"
 echo "Cloning ${URL} in ${TMP_DIR}..."
 git clone --quiet --depth=1 "${URL}" "${TMP_DIR}"
 pushd "${TMP_DIR}" > /dev/null
-go mod init
+go mod init github.com/client9/misspell
```
Comment (chuckha) [inline test/infrastructure/docker/hack/verify-spelling.sh]: we should use hack/tools/tools.go for this

File: Makefile [hunk 2]
```diff
@@ -500,6 +502,11 @@ verify-gen: generate
 		echo "generated files are out of date, run make generate"; exit 1; \
 	fi
 
+.PHONY: capd-verify
+capd-verify:
```
Comment (vincepri) [inline Makefile]:
```suggestion
verify-docker-provider:
```

for consistency with the other targets

File: test/infrastructure/docker/hack/verify-spelling.sh (hunk 1 above)
Comment (ashish-amarnath) [reply test/infrastructure/docker/hack/verify-spelling.sh]: github.com/client9/misspell doesn't use go modules and not seeing a way around that other than using this script.

File: test/infrastructure/docker/hack/verify-spelling.sh [hunk 3]
```diff
@@ -38,7 +38,7 @@ URL="https://github.com/client9/misspell"
 echo "Cloning ${URL} in ${TMP_DIR}..."
```
Comment (ashish-amarnath) [inline test/infrastructure/docker/hack/verify-spelling.sh]: This script can moved to hack/ in capi repo root.

Comment (vincepri) [reply test/infrastructure/docker/hack/verify-spelling.sh]: This shouldn't be needed as a script, we should enable it in golangci-lint https://github.com/golangci/golangci-lint#disabled-by-default-linters--e--enable

Comment (vincepri) [reply test/infrastructure/docker/hack/verify-spelling.sh]: Same probably applies to `gofmt`, `goimports, `govet`, and so on

Comment (ashish-amarnath) [reply test/infrastructure/docker/hack/verify-spelling.sh]: Agree. We've enabled these linters. https://github.com/kubernetes-sigs/cluster-api/blob/master/.golangci.yml
We can delete these scripts.

File: test/e2e/go.mod [hunk 4]
```diff
@@ -5,8 +5,8 @@ go 1.13
 require (
 	github.com/onsi/ginkgo v1.10.3
 	github.com/onsi/gomega v1.7.1
-	k8s.io/api v0.0.0-20191121015604-11707872ac1c
-	k8s.io/apimachinery v0.0.0-20191121015412-41065c7a8c2a
+	k8s.io/api v0.17.0
+	k8s.io/apimachinery v0.17.0
```
Comment (vincepri) [inline test/e2e/go.mod]: These shouldn't be updates yet

Comment (ashish-amarnath) [reply test/e2e/go.mod]: These were a result of `go mod tidy`. Added replace for those packages to pin versions.

**Comment by ashish-amarnath** [comment]
I am not able to figure out what in this PR is causing tests to fail! 🤔

File: test/infrastructure/docker/hack/verify-spelling.sh (hunk 1 above)
Comment (ashish-amarnath) [reply test/infrastructure/docker/hack/verify-spelling.sh]: This script has been deleted in favor of the spelling linter.

**Comment by chuckha** [comment]
/approve

This looks fine to me, thanks for working through all the issues!

**Pull request merged**